src/c++/perf_analyzer/genai-perf/tests/L0_type_checking/test.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+mypy --config-file ./tests/L0_type_checking/mypi.ini
```

Comment:
Purely curiosity - why a test rather than pre-commit/action?

Reply:
thought we were trying to keep precommits uniform across repos?

Reply:
Great suggestion! Done. This PR just grew in scope and I missed it. I was fixing the pylance errors as the team worked on that, then Brian mentioned this is how the MA repo checks for type errors so I modeled it after that. We could probably add it there as well.

Once this PR goes in, I'll let everyone working on GenAi-Perf know to re-run `pre-commit install` to catch any errors.

Reply:
awesome. thank you